### DISCUSSION TITLE
Recommended way to read and write data to notebook cell's metadata

### DISCUSSION THREAD

**Discussion by HatemMn** [microsoft/vscode-discussions#747] [Extension Development QnA] [answered]
As mentioned in the title, my goal is to know which are the best practices in order to change a notebook's metadata as I found no documentation about the matter.
For info, my vscode extension is a notebook renderer. Ideally, I want to write to the cell metadata from within the renderer but I see no clear way of acheiving that.

One thing I have thought of was to send the extension host the data that I wanted to write and then do this : 
```
      const activeNotebookEditor = vscode.window.activeNotebookEditor;
      const noteBouk = activeNotebookEditor.notebook;
      const cell = noteBouk.getCells()[activeNotebookEditor.selection.start]; // I would like to get the active cell
      cell.metadata = { someData: "my_awesome_data" }; // ERROR : Cannot assign to 'metadata' because it is a read-only property
```

If my question is not clear I will be glad to provide more information

**Comment by isidorn** [comment]
I am not sure. Maybe @amunger might be able to help

**Comment by r3m0t** [comment]
I think you need to create and apply a WorkspaceEdit which contains a NotebookEdit.

**Comment by HatemMn** [comment]
Thanks for replies, but I have already tried the WorkspaceEdit and it does not work for this specific task.
An issue is already submitted on the vscode repo but did not get a reply : https://github.com/microsoft/vscode/issues/190160
Also, I did not find documentation about this feature

**Comment by amunger** [comment]
responded in the linked issue, but yes, the edit does need to happen on the extension host.

**Comment by HatemMn** [comment, marked as answer]
To make a summary, the way of solving this problem is using a WorkspaceEdit, `edit = new vscode.WorkspaceEdit();` , setting it and then applying it. Everything is detailed on this issue : https://github.com/microsoft/vscode/issues/190160
Thanks to @amunger for his help !
